src/app/teacher/run-menu/run-menu.component.ts

```diff
@@ -86,4 +83,57 @@ export class RunMenuComponent implements OnInit {
       this.router.navigateByUrl(this.editLink);
     }
   }
+
+  protected archive(archive: boolean): void {
+    this.archiveProjectService[archive ? 'archiveProject' : 'unarchiveProject'](
+      this.run.project
+    ).subscribe({
+      next: (response: ArchiveProjectResponse) => {
+        this.updateArchivedStatus(this.run, response.archived);
+        this.showSuccessMessage(this.run, archive);
+      },
+      error: () => {
+        this.showErrorMessage(archive);
+      }
+    });
+  }
+
+  private showSuccessMessage(run: TeacherRun, archive: boolean): void {
+    this.openSnackBar(
+      run,
+      $localize`Successfully ${archive ? 'archived' : 'restored'} unit.`,
+      archive ? 'unarchiveProject' : 'archiveProject'
+    );
+  }
+
+  private showErrorMessage(archive: boolean): void {
+    this.snackBar.open($localize`Error ${archive ? 'archiving' : 'unarchiving'} unit.`);
+  }
+
+  private updateArchivedStatus(run: TeacherRun, archived: boolean): void {
+    run.archived = archived;
+    this.runArchiveStatusChangedEvent.emit(run);
```

Comment:
Looks like we don't do anything with the emitted value, so make it void?
```suggestion
    this.runArchiveStatusChangedEvent.emit();
```